tests/stash/save.c

```diff
@@ -188,6 +188,46 @@ void test_stash_save__can_include_untracked_and_ignored_files(void)
 	cl_assert(!git_path_exists("stash/just.ignore"));
 }
 
+/*
+ * Note: this test was flaky prior to fixing #4101 -- run it several
+ * times to get a failure.  The issues is that whether the fast
+ * (stat-only) codepath is used inside stash's diff operation depends
+ * on whether files are "racily clean", and there doesn't seem to be
+ * an easy way to force the exact required state.
+ */
+void test_stash_save__untracked_regression(void)
+{
+	git_checkout_options opts = GIT_CHECKOUT_OPTIONS_INIT;
+	const char *paths[] = {"what", "where", "how", "why"};
+	git_reference *head;
+	git_commit *head_commit;
+
+	const char* workdir = git_repository_workdir(repo);
+	size_t len = strlen(workdir);
+	char* untracked_dir = malloc(len + 2);
+
+	strcpy(untracked_dir, workdir);
+	strcpy(untracked_dir + len, "z");
```

Comment:
I'd prefer to just use `git_buf` instead of pointer arithmetic. I mean this here is simple enough and unlikely to break, but `git_buf_printf(&untracked_dir, "%sz", workdir)`s is more readable.